### PR TITLE
Next/20170406/v3

### PR DESCRIPTION
**Pull request by inliniac**
Merge #2633, #2651 and #2652 

Prscript:
- PR inliniac-pcap: https://buildbot.openinfosecfoundation.org/builders/inliniac-pcap/builds/88
- PR inliniac: https://buildbot.openinfosecfoundation.org/builders/inliniac/builds/596

### CODE REVIEW

File: src/util-logopenfile.c [hunk 1]
```diff
@@ -129,37 +180,93 @@ static int SCLogUnixSocketReconnect(LogFileCtx *log_ctx)
...
-            if (SCLogUnixSocketReconnect(log_ctx)) {
-                ret = fwrite(buffer, buffer_len, 1, log_ctx->fp);
-                fflush(log_ctx->fp);
+        /* Check for rotation. */
+        if (log_ctx->rotation_flag) {
+            log_ctx->rotation_flag = 0;
+            SCConfLogReopen(log_ctx);
+        }
+
+        if (log_ctx->flags & LOGFILE_ROTATE_INTERVAL) {
```
Comment (inliniac) [inline src/util-logopenfile.c]: @jasonish @thus I had to solve a merge conflict between your branches here, can you both check that it's correct?

Comment (jasonish) [inline src/util-logopenfile.c]: Look correct.

Comment (thus) [inline src/util-logopenfile.c]: I've tested it, and it works as expected, so it looks good to me.

Comment (inliniac) [inline src/util-logopenfile.c]: Thanks guys!

**Pull request merged**